src/controller/queryPipeline/QueryProcessController.ts

```diff
@@ -0,0 +1,87 @@
+import eventBus from '../EventBus.js';
+import DataStore from '../DataStore.js';
+import { DATA_EVENTS, DRAWING_EVENTS } from '../constants/events.js';
+import { pixelExtractor } from './pixelExtractor.js';
+import { throttle } from './throttle';
+import { IQueryProcessControllerProps } from './types/QueryProcessController.js';
+import {
+    ICanvasBase,
+    IUserInputCanvas,
+    IPathTrackingCanvas,
+    IAlignCanvas,
+    IResizeCanvas,
+} from '@/view/components/userQuery/types/canvas';
+import { INeuralNetworkBase } from '@/core/types/NeuralNetworkBase';
+import { IDataStore } from '../types/DataStore';
+import { Matrix2D } from '@/core/ops/types/OpsType.ts';
+import BoundingBox from '@/view/components/userQuery/canvasUtils/BoundingBox.ts';
+
+class QueryProcessController {
+    private readonly userInputCanvas: ICanvasBase & IUserInputCanvas;
+    private readonly trackingCanvas: ICanvasBase & IPathTrackingCanvas;
+    private readonly alignCanvas: ICanvasBase & IAlignCanvas;
+    private readonly resizeCanvas: ICanvasBase & IResizeCanvas;
+    private readonly $DS: IDataStore;
+    private readonly $NN: INeuralNetworkBase;
+    private readonly queryFrequencyMs: number;
+    //TODO: Canvas clear시 Skeleton 에니메이션 적용하기
+
+    constructor({
+        userInputCanvas,
+        trackingCanvas,
+        alignCanvas,
+        resizeCanvas,
+        $NN,
+    }: IQueryProcessControllerProps) {
+        this.userInputCanvas = userInputCanvas;
+        this.trackingCanvas = trackingCanvas;
+        this.alignCanvas = alignCanvas;
+        this.resizeCanvas = resizeCanvas;
+
+        this.$NN = $NN;
+        this.$DS = DataStore;
+        this.registerDrawingEvent();
+        this.query();
+        this.queryFrequencyMs = 200;
+    }
+
+    registerDrawingEvent(): void {
+        eventBus.on(DRAWING_EVENTS.START_DRAW, ({ x, y }) => {
+            this.userInputCanvas.startPath(x, y);
+            this.trackingCanvas.startPath(x, y);
+        });
+        eventBus.on(DRAWING_EVENTS.DRAW, ({ x, y }) => {
+            this.userInputCanvas.drawPath(x, y);
+            this.trackingCanvas.drawPath(x, y);
+            this.alignCanvas.updateCanvasScale();
+            this.alignCanvas.centralize(this.trackingCanvas.canvas);
+            this.resizeCanvas.downScale(this.alignCanvas.canvas);
+            this.$DS.setQueryInfo(pixelExtractor(this.resizeCanvas));
```

Comment:
캔버스 드로잉을 수행할 때는 [requestAnimationFrame API](https://developer.mozilla.org/ko/docs/Web/API/Window/requestAnimationFrame)를 사용하면 좋아요.

```suggestion
            requestAnimationFrame(() => {
              this.userInputCanvas.drawPath(x, y);
              this.trackingCanvas.drawPath(x, y);
              this.alignCanvas.updateCanvasScale();
              this.alignCanvas.centralize(this.trackingCanvas.canvas);
              this.resizeCanvas.downScale(this.alignCanvas.canvas);
              this.$DS.setQueryInfo(pixelExtractor(this.resizeCanvas));
            })
```

생각해보면 좋아요) DOM Event를 통한 캔버스드로잉 트리거 시점과 실제 캔버스 드로잉 시점이 일치할 필요가 없어요.
(디스플레이 새로고침 주기가 60Hz인데, 실제 캔버스 드로잉이 1초에 120번 발생한다면 성능적으로 어떨까요?)